### PR TITLE
ER-486 QA Notifications

### PR DESCRIPTION
**Pull request by leewadhams**
- Added notifications for vacancy approval and referral.
- Approval/Referral view events are now placed on storage queue and picked up by webjob.
- Ordered reviews on Vacancy Dashboard.

### CODE REVIEW

File: src/Jobs/Recruit.Vacancies.Jobs/VacancyEvents/VacancyEventsJob.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public VacancyEventsJob(ILogger<VacancyEventsJob> logger, VacancyEventHandler ha
             }
             catch (Exception ex)
             {
-                _logger.LogError(ex, $"Unable to handle vacancy event");
+                _logger.LogError(ex, "Unable to handle vacancy event");
```
Comment (chrisjensenuk) [inline src/Jobs/Recruit.Vacancies.Jobs/VacancyEvents/VacancyEventsJob.cs]: Worth logging the message too?

File: src/Jobs/Recruit.Vacancies.Jobs/VacancyReviewEvents/VacancyReviewEventsJob.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            try
+            {
+                var eventItem = JsonConvert.DeserializeObject<EventItem>(message);
+                
+                await UnpackAndExecute(eventItem.EventType, eventItem.Data);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Unable to handle vacancy event");
```
Comment (chrisjensenuk) [inline src/Jobs/Recruit.Vacancies.Jobs/VacancyReviewEvents/VacancyReviewEventsJob.cs]: ditto

File: src/Jobs/Recruit.Vacancies.Jobs/VacancyReviewEvents/VacancyReviewEventsJob.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        private readonly ILogger<VacancyReviewEventsJob> _logger;
+        private readonly VacancyReviewEventHandler _reviewHandler;
+
+        public VacancyReviewEventsJob(ILogger<VacancyReviewEventsJob> logger, VacancyReviewEventHandler reviewHandler)
+        {
+            _logger = logger;
+            _reviewHandler = reviewHandler;
+        }
+
+        public async Task HandleVacancyEvent([QueueTrigger(QueueNames.VacancyReviewEventsQueueName, Connection = "EventQueueConnectionString")] string message, TextWriter log)
```
Comment (chrisjensenuk) [inline src/Jobs/Recruit.Vacancies.Jobs/VacancyReviewEvents/VacancyReviewEventsJob.cs]: HandleVacancyReviewEvent?

File: src/Shared/Recruit.Vacancies.Client/Infrastructure/Events/StorageQueueEventQueue.cs [hunk 4]
```diff
@@ -45,8 +45,13 @@ public StorageQueueEventQueue(StorageQueueConnectionDetails details)
 
         private static string GetQueueName(IEvent @event)
         {
-            if (@event is IVacancyEvent)
-                return QueueNames.VacancyEventsQueueName;
+            switch (@event)
+            {
+                case IVacancyEvent _:
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Vacancies.Client/Infrastructure/Events/StorageQueueEventQueue.cs]: 😮 👍

File: src/Jobs/Recruit.Vacancies.Jobs/VacancyEvents/VacancyEventsJob.cs (hunk 1 above)
Comment (leewadhams) [reply src/Jobs/Recruit.Vacancies.Jobs/VacancyEvents/VacancyEventsJob.cs]: Will pick this up in the logging story.

File: src/Jobs/Recruit.Vacancies.Jobs/VacancyReviewEvents/VacancyReviewEventsJob.cs (hunk 3 above)
Comment (leewadhams) [reply src/Jobs/Recruit.Vacancies.Jobs/VacancyReviewEvents/VacancyReviewEventsJob.cs]: 👍

**Pull request merged**